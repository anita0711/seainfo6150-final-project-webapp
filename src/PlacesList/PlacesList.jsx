import React from "react";
import { Link } from "react-router-dom";
import PlaceListItem from "../PlaceListItem/PlaceListItem";
import styles from "./PlacesList.module.css";

const PlacesList = (props) => {
  return (
    <div>
      <div className={styles.heading}>{props.pageTitle}</div>
      <ul className={styles.placesList}>
        {props.places.map((place, index) => (
          <Link to={`/place/${place.category}/${place.placeId}`}>
            <li key={index}>
              <PlaceListItem img1={place.img1._url} title={place.title} />
            </li>
          </Link>
        ))}
      </ul>
    </div>
  );
};

export default PlacesList;
