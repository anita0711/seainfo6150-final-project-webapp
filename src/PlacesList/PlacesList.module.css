.placesList {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  list-style: none;
  column-gap: 25px;
  row-gap: 25px;
  padding-left: 10px;
}

.title {
  display: inline-block;
  background-color: #ccc;
  border-radius: 10px;
  margin-bottom: 15px;
  margin-left: 10px;
  margin-top: 10px;
}

@media screen and (min-width: 150px) {
  .placesList {
    grid-template-columns: 1fr;
  }
}

@media screen and (min-width: 500px) and (max-width: 800px) {
  .placesList {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media screen and (min-width: 800px) {
  .placesList {
    grid-template-columns: repeat(3, 1fr);
  }
}
