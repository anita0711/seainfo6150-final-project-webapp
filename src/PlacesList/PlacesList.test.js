import React from "react";
import { render } from "@testing-library/react";
import PlacesList from "./PlacesList.jsx";
import { BrowserRouter } from "react-router-dom";

it("renders the PlacesList component correctly", () => {
  const placesTestList = [
    {
      category: "summer",
      placeId: "001-yellowstone",
      state: "Wyoming",
      title: "Yellowstone",
      img1: {
        _url:
          "https://www.planetware.com/photos-large/USWY/usa-wyoming-yellowstone-grand-prismatic-spring.jpg",
      },
    },
    {
      category: "summer",
      placeId: "004-glacierNationalPark",
      state: "Montana",
      title: "Glacier National Park",
      img1: {
        _url:
          "https://www.doi.gov/sites/doi.gov/files/blog-post/thumbnail-images/glaciernpshanlin.jpg",
      },
    },
  ];
  const { container } = render(
    <BrowserRouter>
      <PlacesList
        books={Object.values(placesTestList)}
        pageTitle="Best places for Summer season"
      />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
