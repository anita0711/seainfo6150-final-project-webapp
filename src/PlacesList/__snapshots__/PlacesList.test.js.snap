// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PlacesList component correctly 1`] = `
<div>
  <div>
    <div
      class="title"
    >
      <h3>
        Best places for Summer season
      </h3>
    </div>
    <ul
      class="placesList"
    >
      <li
        class="placeLink"
      >
        <a
          href="/place/summer/001-yellowstone"
        >
          <div
            class="placeListItem"
          >
            <img
              alt="Yellowstone"
              class="image"
              src="https://www.planetware.com/photos-large/USWY/usa-wyoming-yellowstone-grand-prismatic-spring.jpg"
            />
            <p
              class="placeName"
            >
              Yellowstone
            </p>
          </div>
        </a>
      </li>
      <li
        class="placeLink"
      >
        <a
          href="/place/summer/004-glacierNationalPark"
        >
          <div
            class="placeListItem"
          >
            <img
              alt="Glacier National Park"
              class="image"
              src="https://www.doi.gov/sites/doi.gov/files/blog-post/thumbnail-images/glaciernpshanlin.jpg"
            />
            <p
              class="placeName"
            >
              Glacier National Park
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
